go-controller/pkg/ovn/master.go

```diff
@@ -114,36 +116,49 @@ func (oc *DefaultNetworkController) upgradeOVNTopology(existingNodes *kapi.NodeL
 	return err
 }
 
-// SetupMaster creates the central router and load-balancers for the network
-func (oc *DefaultNetworkController) SetupMaster(existingNodeNames []string) error {
+// createOvnClusterRouter creates the central router for the network
+func (bnc *BaseNetworkController) createOvnClusterRouter(multicastSupport bool) (*nbdb.LogicalRouter, error) {
```

Comment:
I wonder if this should be a receiver func of the base controller; why would a network controller for a flat L2 secondary network know how to do this ?

Reply:
because we didn't introduce the secondarylayer3controller, and NetworkControllerInfo structure yet. That would have more information we need to indicate what exactly needs to be done.